### ISSUE TITLE
Latest changes regarding multiple JWK URLs are now breaking the old logic

### ISSUE THREAD

**Issue by sherifkayad** [Robbilie/nginx-jwt-auth#2]
hi, I am using Cognito which returns multiple JWKs in its endpoint. That used to work in the past, however, not anymore with the commit introduced here https://github.com/Robbilie/nginx-jwt-auth/commit/9a62df22bd6ea676703c4faae2c9eaf1aea74a60

Any clue how one could get arround that?

Cognito responds to the `/.well-known/jwks.json` with something like below:
![image](https://github.com/Robbilie/nginx-jwt-auth/assets/8362784/12ee6d89-ca43-4ced-b87e-fdbf64bdf23c)

Now I am getting the error below:
```
{"level":"fatal","ts":"2023-05-11T15:27:07.181Z","caller":"src/main.go:64","msg":"Couldn't initialize server","err":"failed to create JWKS from resource at the given URL.\nError: multiple JWKS must have two or more remote JWK Set resources: multiple JWKS must have two or more remote JWK Set resources"}
```

**Comment by sherifkayad** [comment]
@Robbilie I checked out the code locally and traced to the lib `MicahParks/keyfunc` .. The problem with the library used is that the function `keyFunc.getMultiple` is expecting a map with at least two entries 😄 .. Check this line here https://github.com/MicahParks/keyfunc/blob/master/multiple.go#L25

A simple and yet dumb workaround is provide the `JWKS_URL` as follows: `https://cognito-idp.bla.blob/.well-known/jwks.json,https://cognito-idp.bla.blob/.well-known/jwks.json?ping=pong`

**Comment by sherifkayad** [comment]
So the problem isn't that Cognito is returning two keys .. I think the nginx-jwt-auth need to differentiate between using 1 or multiple JWKS endpoints and based on that either use `keyFunc.getMultiple` or `keyfunc.Get`

**Comment by Robbilie** [comment]
yeah im nearly done with a fix, give me a few minutes :P

**Comment by sherifkayad** [comment]
you are awesome 😄

**Comment by Robbilie** [comment]
i tested it locally with one url, 2 urls, 2x the same url. the last one still returns the same error but i would say this is tolerable, because that would be considered user error :D

i hope the refresh interval still works properly but theres not really a reason not to…

**Comment by Robbilie** [comment]
https://github.com/Robbilie/nginx-jwt-auth/commit/a24e2f3b8dc493ff42b99e043ca4ca45940d3274

**Comment by sherifkayad** [comment]
let pull the image when it's built and will let you know ;) .. BTW would be great if you can issue a release once we make sure it's working fine. What do you think?

I would love to use a pinned version of your changes :P

**Comment by Robbilie** [comment]
i already created a release, let me know if it works :)
(i personally also run the pinned version v0.0.4 until now thats why i didnt hit this issue yet)

**Comment by sherifkayad** [comment]
awesome! works like a charm .. Thanks a lot for your prompt response ;) looking forward to the new release .. I will pin it in my configs then.

**Comment by Robbilie** [comment]
v0.0.5 is available as an image with the tag 0.0.5
ghcr.io/robbilie/nginx-jwt-auth:0.0.5
